sklearn/cluster/affinity_propagation_.py

```diff
@@ -288,6 +302,12 @@ def fit(self, X, y=None):
         if self.affinity == "precomputed":
             self.affinity_matrix_ = X
         elif self.affinity == "euclidean":
+            warnings.warn("Using squared distances. In the future please use " 
+                          "sqeuclidean for squared and euclidean for "
+                          "for non-squared distances.",
+                          DeprecationWarning)
+            self.affinity_matrix_ = -euclidean_distances(X, squared=False)
```

Comment:
this is not the right fix. You'll get a warning all the time even if you know what you're doing ie. want actually "euclidean"

this is also not backward compatible.